bionic/deriver.py

```diff
@@ -88,11 +97,31 @@ def export_dag(self, include_core=False):
         """
         nx = import_optional_dependency("networkx", purpose="constructing the flow DAG")
 
-        def should_include_dnode(dnode):
-            if include_core:
-                return True
+        # We only want to include certain nodes in the visualization: generally the ones
+        # corresponding to user-defined functions and/or entities. (These are usually
+        # the same, but not when a user function returns a more complex descriptor.) For
+        # other nodes, we "collapse" them: we delete the node and connect each of its
+        # parents to each of its children.
+        def should_collapse_dnode(dnode):
+            if not include_core:
+                descriptor_includes_non_core = any(
+                    not entity_is_internal(entity_name)
+                    for entity_name in dnode.all_entity_names()
+                )
+                if not descriptor_includes_non_core:
```

Comment:
This means that all entities are core (or internal). I think you can simplify this by using `all` and removing `not`s (both inside `any` and outside here).

Reply:
You're right, this looks nicer!